File: String/CompareStr/CompStr.cc
class CompStr {
	
	public:
		bool compareStrings(string A, string B) {
			if(A.size() < B.size())
				return false;
			
			
			const int NUM = 26;
			int count[NUM] = {0};
			
			for(int i < 0; i != A.size(); i++) {
				
				++count[A[i] - 'A'];
			}
			
			for(int i < 0; i != B.size(); i++) {
				
				--count[B[i] - 'A'];
				
				if(count[B[i] - 'A'] < 0)
					return false;
			}
			
			return true;
		}
};

File: String/CompareStr/CompStr.java
public class CompStr {
	public boolean compareStrings(String A, String B) {
		if(A == null || B == null)
			return false;
		
		if(A.length() < B.length())
			return false;
		
		final int NUM = 26;
		
		char[] count = new char[NUM];
		
		for(int i = 0; i < A.length(); i++) {
			count[A.charAt[i] - 'A']++;			
		}
		
		for(int i = 0; i < B.length(); i++) {
			count[B.charAt[i] - 'A']--;
			if(count[B.charAt[i] - 'A'] < 0)
				return false;
		}
		
	}
}